flavors/vanilla/12/default.nix

```diff
@@ -103,10 +103,10 @@ in
         url = "https://github.com/aosp-mirror/platform_frameworks_base/commit/0856f76846e61ad058e1e9ec0759739812a00600.patch";
         sha256 = "sha256-of8dyOCicSVh64kLicuIk9av2K29YXLzUxeb6CI0NZo=";
       })
-      (pkgs.fetchurl {
+      (pkgs.fetchpatch {
         name = "include-saved-battery-history-chunks-into-batteryusagestats-parcel.patch";
         url = "https://github.com/aosp-mirror/platform_frameworks_base/commit/c4b9de7d95fd2d6bd8072f16f0ac71d2b1773a1b.patch";
-        sha256 = "sha256-ZIi96yF2qTgQ4iGTY86ppBmg4TeIRJ1qu7CSA5IPSnE=";
+        sha256 = "sha256-T9EBOMJJtZ2tPOYx3epXV9ZBg3gSuz3zwM7+9QIOgs8=";
```

Comment:
You'd also need to update this hash again after you've switched to `fetchpatch`. I get this hash when I try to fetch:
`sha256-SalnJQSrc7X24XW1vg0N04PYIQ11vm++xy/CPbPogy4=`

Reply:
Thanks. I forgot about that annoying but necessary dance of changing the hash to something different and then changing it to the ACTUALLY correct one instead of the original one which just looks like it's working.